Added Getting Started guide for TFLite (#8136)
google/iree: docs/website/docs/getting-started/index.md

> # Getting Started Guide

Author: I agree overall. My thought was to rewrite entirely with a better structure, explicit examples, etc. Then remove the ml-frameworks sections. Rewording to use the term ml-frameworks seems like a good idea but retrofitting the existing docs feels tricky. It felt easier to rewrite / restructure entirely then remove.

> # Getting Started Guide

Reviewer: Sure, let's try to land any rewrites in a single step though, IMO. I'd rather not have a set of deprecated docs live on the website. I still haven't cleaned up / deleted the even older docs at https://github.com/google/iree/tree/main/docs/developers/get_started :P

> # Getting Started Guide

Author: Sounds good.

> ## Setup

Author: Done. Can test review now.

To setup your local environment for IREE it is recommended to use our uploaded binaries
rather than building the entire project. There is a comprehensive installation 

> [guide](../bindings/python.md), to install all available front-ends execute the following:

Author: Reworded entirely to remove person and be more succinct.

```
python -m pip install \
  iree-compiler \
  iree-runtime \
  iree-tools-tf \
  iree-tools-tflite \
  iree-tools-xla \

>   --find-links https://github.com/google/iree/releases

Author: Done.

```

## Examples

We include simple end-to-end examples of executing an existing model on IREE. This includes
the import, compilation, and final test execution of the provided model. For examples select
a link below.

[TensorFlow Lite](./tflite/index.md)

The remaining details will outline the tasked performed by each stage.

## Import

Importing the model takes a known filetype and imports it into a form that the core IREE
compiler is able to ingest. This import process is specific per-frontend and typically

> run a couple stages:

Author: Done.

* Load the source format
* Legalize operations specific for specific frontend to a known IR
* Validate only legal operations remain
* The resulting file is written to an MLIR containing IREE supported operations

This fully legalized form can then be compiled without depending on the model's original format.

## Compilation

> During compilation we load an MLIR file and compile for a set of specified backends (cpu, gpu, etc).

Author: Done.

Each of these backends creates custom native code to execute on the target device.  Once compiled,
the resulting bytecode is exported to an IREE bytecode file that can be executed on the specified devices.

## Execution

The final stage is executing the now compiled module. This involves selecting what compute devices
should be used, loading the module, and executing the module with the intended inputs. For testing

> IREE includes a Python API however on smaller devices you will want to use the C/C++ APIs.

Author: Done.

# Getting Started Guide

## Setup
Setup your local environment by installing IREE's prebuilt binaries. See the comprehensive

> installation [guide](../bindings/python.md) for installation of individual components, the default installation is below.

Reviewer 2: "Use the following command for the default installation, or check out the comprehensive installation guide if your needs are more complex.

Author: Done.

```
python -m pip install \
  iree-compiler \
  iree-runtime \
  iree-tools-tf \
  iree-tools-tflite \
  iree-tools-xla
```

## Supported frameworks

See end-to-end examples of how to execute a variety models on IREE. This covers
the import, compilation, and execution of the provided model.

* [TensorFlow](./tensorflow.md)
* [TensorFlow Lite](./tflite.md)
* [JAX](./jax.md)

Importing from PyTorch and other frameworks is planned - stay tuned!

## Samples

Check out the samples in IREE's

> [colab/ directory](https://github.com/google/iree/tree/main/colab) and the

Reviewer 2: Check out the samples in IREE's colab/ directory, as well as the iree-samples respository, which contains workflow comparisons across frameworks.

Author: Done.

[iree-samples repository](https://github.com/google/iree-samples) for examples
and workflow comparisons across frameworks.

## Import

Importing models takes known file types and imports into a form that the core IREE
compiler is able to ingest. This import process is specific to each frontend and typically

> runs a couple of stages:

Reviewer 2: ```suggestion
involves a number of stages:
```

* Load the source format
* Legalize operations specific for specific frontend to a known IR
* Validate only legal operations remain

> * Write the remaining IR to an IREE compatible format

Reviewer 2: ```suggestion
* Write the remaining IR to an IREE-compatible format
```

> This fully legalized form can then be compiled without depending on the 

Reviewer 2: is this the "IREE-compatible format" above? I suggest connecting the dots more explicitly

Author: Updated to make contents clearer.

original model format.

## Compilation

> During compilation we load an MLIR file and compile for a set of specified backends

Reviewer 2: ```suggestion
During compilation we load an MLIR file and compile for the specified set of backends
```

(CPU, GPU, etc).  Each of these backends creates custom native code to execute on the
target device.  Once compiled, the resulting bytecode is exported to an IREE bytecode
file that can be executed on the specified devices.

## Execution

The final stage is executing the now compiled module. This involves selecting what
compute devices should be used, loading the module, and executing the module with the

> intended inputs. For testing IREE includes a Python API however on smaller devices you

Reviewer 2: ```suggestion
intended inputs. For testing, IREE includes a Python API however on smaller devices you
```

> ## Execution
> 
> The final stage is executing the now compiled module. This involves selecting what
> compute devices should be used, loading the module, and executing the module with the
> intended inputs. For testing IREE includes a Python API however on smaller devices you
> will want to use the [C API](../deployment-configurations/index.md).

Reviewer: let's reword this - the choice of binding (of which the C API is one) and deployment configuration involves more than just "Python" vs "small device"

```suggestion
## Execution

The final stage is executing the now compiled module. This involves selecting what
compute devices should be used, loading the module, and executing the module with the
intended inputs. IREE includes a runtime Python API for testing, however for more
complex scenarios you will want to review IREE's supported
[deployment configurations](../deployment-configurations/index.md) and
[bindings](../bindings/index.md).
```
